blogs/2023/07/20/mangling-vscode.md

```diff
@@ -0,0 +1,177 @@
+
+The export mangling work shipped last iteration, further reducing the size of `workbench.js` from 10.6MB to 9.8MB. All reductions in total, this file is now 20% smaller than it would be without mangling. Across all of VS Code, mangling removes 3.9MB of JavaScript code from our compiled sources. Not only is that a nice reduction in download size and install size, that's also 3.9MB less JavaScript that needs to be scanned every single time you start VS Code.
+
+This chart shows the size of `workbench.js` over time. Notice the two drops on the right side. The first big drop in VS Code 1.74 is the result of mangling private properties. The second smaller drop in 1.80 is from mangling exports.
+
+![Zoomed in chart showing the drops from mangling](code-size-after-zoomed.png)
+
+![The size of 'workbench.js' over all VS Code releases, including the mangling work](code-size-after.png)
+
+Our mangling implementation can doubtless be improved since our minified sources still contains plenty of long names. We may investigate these further if doing so seems worthwhile and if we can come up with a safe approach. Ideally, some day much of this work won't be necessary at all. Native private properties are already mangled automatically and our build tools will hopefully become better at optimizing code across our entire codebase. You can review our current [mangling implementation](https://github.com/microsoft/vscode/blob/48cd8e0c1b142a46f0956b593d8331145634658e/build/lib/mangle/index.ts).
+
+We're always striving to make VS Code and our codebase better, and I think the mangling work is a great demonstration of how we approach this. Optimization is an ongoing process, not a one time thing. By continually monitoring our code size, we were aware of how it has grown over time. This awareness has doubtless help keep our code size from expanding even more than it has, and also encourages us to always be looking for improvements. Although mangling was an attractive seeming technique, it was initially too risky to seriously consider. Only once we had worked to reduce this risk, create the right safety nets, and make the cost of adopting mangling almost zero, did we finally feel confident enough to enable it in our builds. I'm really proud of the end result and just as proud of how we went about achieving it.
+
+Happy Coding,
+
+Matt Bierner, VS Code Team Member
+[@mattbierner](https://twitter.com/mattbierner)
+
+---
+
+Thank you to Johannes Rieken for his key work implementing mangling, to the TypeScript team for building the tools that let us implement mangling safely, to esbuild for their blazingly fast bundler, and to the entire VS Code team for building a codebase that is fit for optimizations like this. And last but certainly not least, a huge thanks to the V8 team for always making us look fast despite the heaps and heaps of horribly mangled JavaScript we throw their way.
```

Comment:
Should generalise the "Thank You" to JS engines of all browsers because of vscode.dev?